### PR TITLE
#9288: Add ttnn support for distilbert model

### PR DESCRIPTION
**Pull request by Sudharsan-V**
### Ticket
- Link to Github Issue [#9288](https://github.com/tenstorrent/tt-metal/issues/9288)

### Problem description
- In this PR, the support for Distilbert Question Answering model is enabled using ttnn ops.
- ttnn do not have support for masked_fill op, hence pytorch masked_fill op is used.
- The files are organised in the `models/experimental` folder as the model uses pytorch op

### Checklist
- [x] Post commit CI passes [all post commit](https://github.com/tenstorrent/tt-metal/actions/runs/10805524451)
- [x] Model regression CI testing passes (if applicable) [Model perf](https://github.com/tenstorrent/tt-metal/actions/runs/10805519011)
- [ ] New/Existing tests provide coverage for changes
- [x] Device Perf Regression [Device Perf](https://github.com/tenstorrent/tt-metal/actions/runs/10805529028)

### CODE REVIEW

File: models/experimental/functional_distilbert/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+## DIstilbert Model
+
```
Comment (tt-rkim) [inline models/experimental/functional_distilbert/README.md:2]: Please clarify what platforms (single card GS E150, WH N150, WH N300, T3K etc) this model is meant work for.

Customers with various systems will come and expect us to tell them what does and does not work.

File: models/experimental/functional_distilbert/tests/test_perf_distilbert.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+
+@pytest.mark.models_device_performance_bare_metal
+@pytest.mark.parametrize(
+    "batch_size, test, expected_perf",
+    [
+        [8, "384-8-phiyodr/bert-large-finetuned-squad2", 260],
+    ],
+)
+def test_distilbert_perf_device(batch_size, test, expected_perf, reset_seeds):
```
Comment (tt-rkim) [inline models/experimental/functional_distilbert/tests/test_perf_distilbert.py:140]: Why would you not add this to the device perf suite?

Comment (tt-rkim) [reply models/experimental/functional_distilbert/tests/test_perf_distilbert.py:140]: You used the `models_device_performance_bare_metal` with the expected perf as a device cycles count. This means you mean to add it to the device perf suite: https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml

However, I do not see this test run in the section of the performance script for device performance.

File: models/experimental/functional_distilbert/README.md (hunk 1 above)
Comment (Sudharsan-V) [reply models/experimental/functional_distilbert/README.md:2]: Sure, will check it

File: models/experimental/functional_distilbert/tests/test_perf_distilbert.py (hunk 2 above)
Comment (Sudharsan-V) [reply models/experimental/functional_distilbert/tests/test_perf_distilbert.py:140]: Will add this test to device_perf suite.

**Comment by tt-rkim** [comment]
Please post passing links for both

e2e perf
device perf

for single card and show exactly where it runs in the log. Then I will approve

**Review by arakhmati — APPROVED** [review]

**Review by TT-BrianLiu — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Comment by Sudharsan-V** [comment]
> Please post passing links for both
> 
> e2e perf device perf
> 
> for single card and show exactly where it runs in the log. Then I will approve

@tt-rkim , Please find the links for the Device and Model perf below:

- Device Perf [link](https://github.com/tenstorrent/tt-metal/actions/runs/10055463474)
- Model Perf [link](https://github.com/tenstorrent/tt-metal/actions/runs/10057442802)

**Review by tt-rkim — APPROVED** [review]

File: models/experimental/functional_distilbert/README.md (hunk 1 above)
Comment (tt-rkim) [reply models/experimental/functional_distilbert/README.md:2]: This doesn't seem done, still?

**Comment by tt-rkim** [comment]
Any update on:

- ownership information in the README
- whether ttnn integration test passes on N150/N300?

File: models/experimental/functional_distilbert/README.md (hunk 1 above)
Comment (Sudharsan-V) [reply models/experimental/functional_distilbert/README.md:2]: Updated the platforms on which the model works.

**Comment by Sudharsan-V** [comment]
> Any update on:
> 
> * ownership information in the README
> * whether ttnn integration test passes on N150/N300?

1. The README is updated with the Owner's name.
2. The tests pass on N150 and N300.

**Pull request merged**